Sources/CSRF/CSRF.swift

```diff
@@ -42,30 +42,29 @@ public struct CSRF: Middleware {
     /// Creates a token from a given `Request`. Call this method to generate a CSRF token to assign to your key of choice in the header and pass the token back to the caller via the response.
     /// - parameter request: The `Request` used to either find the secret in, or the request used to generate the secret.
     /// - returns: `Bytes` representing the generated token.
-    /// - throws: An error that may arise from either creating the secret from the request or from generating the token.
-    public func createToken(from request: Request) -> String {
+    public static func createToken(from request: Request) -> String {
         let secret = createSecret(from: request)
         let saltBytes = [UInt8].random(count: 8)
         let saltString = saltBytes.description
         return generateToken(from: secret, with: saltString)
     }
     
-    private func generateToken(from secret: String, with salt: String) -> String {
+    private static func generateToken(from secret: String, with salt: String) -> String {
         let saltPlusSecret = (salt + "-" + secret)
         let digest = Insecure.MD5.hash(data: [UInt8](saltPlusSecret.utf8))
         let token = digest.description
         return salt + "-" + token
     }
     
-    private func validate(_ token: String, with secret: String) throws -> Bool {
+    private static func validate(_ token: String, with secret: String) throws -> Bool {
```

Comment:
Why was this (and the other methods) made `static`?

Reply:
Those functions do not access any fields of the instance. So they don't need to be instance methods but can be class functions instead. IMHO that leads to cleaner code.

It also makes the usage of CSRF easier
a) when using it in code
b) in the CSRFFormFieldTag as it would probably be a pain to access the middleware just from the request object.

But as I said, the used function doesn't need access to anything of the instance and therefore can be static…

Reply:
Fair enough!